### PR TITLE
Add ExternalAccess entrypoint for removing todocomments

### PR DESCRIPTION
**Pull request by uniqueiniquity**
Currently, if a JS/TS file with todo comments is deleted, the corresponding comments aren't removed from the task list. This is because Roslyn OOP isn't tracking those files, whereas we are.

To handle this, this PR adds an entry point so we can report when documents we track are deleted.

### CODE REVIEW

**Comment by uniqueiniquity** [comment]
cc: @CyrusNajmabadi

File: src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs [hunk 1]
```diff
@@ -221,5 +221,9 @@ public async Task ReportTodoCommentDataAsync(DocumentId documentId, ImmutableArr
             await ReportTodoCommentDataAsync(
                 document.Id, converted.ToImmutable(), cancellationToken).ConfigureAwait(false);
         }
+
+        /// <inheritdoc cref="IVsTypeScriptTodoCommentService.OnDocumentRemovedAsync(DocumentId, CancellationToken)"/>
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs:225]: 📝 This is automatic if you want to remove it

Comment (uniqueiniquity) [reply src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs]: Removed here and above. Thanks!

**Comment by uniqueiniquity** [comment]
On further inspection, I've filed https://github.com/dotnet/roslyn/issues/44062 to track that the issue also repros for C# files. I imagine we'll still need this API once it works correctly, but we'll have to see.

File: src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs [hunk 2]
```diff
@@ -221,5 +220,8 @@ public async Task ReportTodoCommentDataAsync(DocumentId documentId, ImmutableArr
             await ReportTodoCommentDataAsync(
                 document.Id, converted.ToImmutable(), cancellationToken).ConfigureAwait(false);
         }
+
+        Task IVsTypeScriptTodoCommentService.OnDocumentRemovedAsync(DocumentId documentId, CancellationToken cancellationToken) =>
+            OnDocumentRemovedAsync(documentId, cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs:225]:
```suggestion
        Task IVsTypeScriptTodoCommentService.OnDocumentRemovedAsync(DocumentId documentId, CancellationToken cancellationToken)
            => OnDocumentRemovedAsync(documentId, cancellationToken);
```

**Comment by uniqueiniquity** [comment]
Closing in favor of #44274.

**Pull request closed without merging**